lamindb/models/artifact.py

```diff
@@ -2142,23 +2148,30 @@ def replace(
         self._old_suffix = self.suffix
 
     def open(
-        self, mode: str = "r", is_run_input: bool | None = None, **kwargs
-    ) -> Union[
-        AnnDataAccessor,
-        BackedAccessor,
-        SOMACollection,
-        SOMAExperiment,
-        SOMAMeasurement,
-        PyArrowDataset,
-    ]:
+        self,
+        mode: str = "r",
+        df_engine: Literal["pyarrow", "polars"] = "pyarrow",
+        is_run_input: bool | None = None,
+        **kwargs,
+    ) -> (
+        AnnDataAccessor
+        | BackedAccessor
+        | SOMACollection
+        | SOMAExperiment
+        | SOMAMeasurement
+        | PyArrowDataset
+        | Iterator[PolarsLazyFrame]
+    ):
         """Return a cloud-backed data object.
 
         Works for `AnnData` (`.h5ad` and `.zarr`), generic `hdf5` and `zarr`,
-        `tiledbsoma` objects (`.tiledbsoma`), `pyarrow` compatible formats.
+        `tiledbsoma` objects (`.tiledbsoma`), `pyarrow` or `polars` compatible formats.
 
         Args:
             mode: can only be `"w"` (write mode) for `tiledbsoma` stores,
                 otherwise should be always `"r"` (read-only mode).
+            df_engine: Which module to use for lazy loading of a dataframe
```

Comment:
Also rename the parameter here, please.

Reply:
I'd make this `engine: Literal["pyarrow", "polars"] | None = None` and then select `pyarrow` by default if it's not passed and it's a `DataFrame`.

It'd also be really nice to add a 2nd example, so that people see this doesn't just work for `AnnData`.

<img width="778" alt="image" src="https://github.com/user-attachments/assets/657147b5-9629-4c48-934d-8ba3cbe144be" />

Reply:
Also should be called "see guide":
<img width="748" alt="image" src="https://github.com/user-attachments/assets/01ec96dc-0b1a-4d87-b371-86cfd0ccf7da" />

Reply:
Edited, added an example.
I actually hate such defaults with `None`, you have to read the description etc to understand what is actually the default value, i believe it is much more obvious with just setting it directly.

Reply:
But you can't if it's called df_engine, because it might then default to something else for another data format.

Reply:
Renamed.